cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -86,14 +105,16 @@ public DatasetInstanceService(DatasetTypeManager typeManager, DatasetInstanceMan
     this.metaCache = CacheBuilder.newBuilder().build(
       new CacheLoader<Id.DatasetInstance, DatasetMeta>() {
         @Override
-        public DatasetMeta load(Id.DatasetInstance datasetId) throws Exception {
+        public DatasetMeta load(@SuppressWarnings("NullableProblems") Id.DatasetInstance datasetId) throws Exception {
           return getFromMds(datasetId);
         }
       }
     );
+    this.authorizationEnforcementService = authorizationEnforcementService;
+    this.authorizerInstantiator = authorizerInstantiator;
   }
 
-  @SuppressWarnings("unused")
+  @VisibleForTesting
   @Inject(optional = true)
   public void setAuditPublisher(AuditPublisher auditPublisher) {
```

Comment:
no, its unfortunate. The initialization in ``RemoteDatasetFrameworkTest`` and ``DatasetServiceTestBase`` needs to be completely rewritten to use guice. Right now it just 'new's everything. I've done it partially in my PRs, but we'll need to do a more thorough refactor (leaving that out to beyond 3.5). So yes, it needs to be public because ``RemoteDatasetFrameworkTest`` calls it.